Makefile

```diff
@@ -21,6 +21,9 @@ interrogate:
 lint:
 	pre-commit run markdownlint -a
 
+ruff:
```

Comment:
Can we call ruff a linter? The make file options correspond to CI stages. We can make a ruff stage is we want but then all repos using the CI pipline would need to add ruff in their make file.

Reply:
* If you think ruff really needs to be a seperate stage, then we totally can. Just a little more work (although I would like to make the stages more generic such as black -> format and interogate -> doctest  or something

Reply:
I was wondering about this. I think we can call ruff a linter and add it to the CI stage.